src/hoc/withEditing/withEditing.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import * as _ from 'underscore';
+import {StickyFooter} from '../../components/stickyFooter/StickyFooter';
+import * as styles from '../../components/stickyFooter/StickyFooter.scss';
+import {IReactVaporState} from '../../ReactVapor';
+import {callIfDefined} from '../../utils/FalsyValuesUtils';
+import {IDispatch, ReduxConnect} from '../../utils/ReduxUtils';
+import {toggleDirtyComponent} from './withEditingActions';
+
+export interface IWithEditing {
+    id: string;
+    isDirty?: boolean;
+    footerChildren?: React.ReactNode;
+}
+
+export interface IWithEditStateProps {
+    isDirty: boolean;
+}
+
+export interface IWithEditDispatchProps {
+    toggleDirtyComponent: (isDirty: boolean) => void;
+}
+
+export const withEditing = (config: IWithEditing) => (Component: React.ComponentClass<any, any>): React.ComponentClass<any, any> => {
+    const mapStateToProps = (state: IReactVaporState): IWithEditStateProps => ({
+        isDirty: !!_.find(state.dirtyComponents, (id) => id === config.id),

```

Comment:
thanks, not sure this is really necessary, we'll rarely have more than one (2,3?) dirty components at the same time, not sure it will ever be expensive really 🤔

Reply:
You're right, there won't be many components at the same time. Although a selector (doesn't have to be a `reselect` one, it could be a simple exported function),  would allow to reuse the logic elsewhere if needed.